### PR TITLE
Remove the SSE 4.2 instruction set from the hardware requirements (`80.x` release branch PR)

### LINKED ISSUE #2955: SSE 4.2 listed as hardware requirement but does not appear to be used
### Are you using the latest Dosbox-Staging Version?

- [X] I have checked releases and am using the latest release.

### Different version than latest?

_No response_

### What Operating System are you using?

Linux x86_64

### If Other OS, please describe

_No response_

### Relevant hardware info

_No response_

### Have you checked that no other similar issue already exists?

- [X] I have searched and not found similar issues.

### A clear and concise description of what the bug is.

On the website, SSE 4.2 is listed as a requirement on x86:  https://dosbox-staging.github.io/downloads/linux/#hardware-requirements

However, I don't think we're setting the correct compiler flags to even utilize this.  We need to be setting `-march` to whatever the minimum supported CPU is.  https://gcc.gnu.org/onlinedocs/gcc/x86-Options.html

Doing a `git grep _m128` (to look for SIMD intrinsics) shows Voodoo emulation doing some SSE2 stuff.  SSE2 should be fine with the default `march` as I believe every x86_64 CPU supports SSE2 at a minimum.

The only other place I'm seeing intrinsics is inside `src/libs`.  `dr_flac.h` (part of SDL Sound from the looks) has a lot of it in particular, including SSE4.1.  However, it's all behind `#ifdefs` that I don't think we're defining anywhere.  I wouldn't be surprised if this is getting built without SIMD at all.  It looks like these files just got copy-pasted and we're compiling without running the build system they were meant to be built with.  The `meson.build` file in that directory is very bare bones.

I noticed this because I was trying some SIMD for the video scaler and the SSE4 intrinsics wouldn't compile without adding `-march=native` to the meson build flags manually.  I'm getting close to 10x the performance with this so I think it's worth getting these compile flags right :smile: 

### Steps to reproduce the behaviour.

_No response_

### Download URL of affected game or software

_No response_

### Your configuration

_No response_

### Provide a Log

_No response_

### Code of Conduct & Contributing Guidelines

- [X] Yes, I agree.

### PR DESCRIPTION
**Pull request by kcgen**
# Description

Remove the SSE 4.2 instruction set from the 80.1-release documentation's hardware requirements.

## Related issues

Fixes #2955 

# Manual testing

# Checklist

_Please tick the items as you have addressed them. Don't remove items; leave the ones that are not applicable unchecked._

I have:

- [x] followed the project's [contributing guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/master/CONTRIBUTING.md) and [code of conduct](https://github.com/dosbox-staging/dosbox-staging/blob/master/CODE_OF_CONDUCT.md).
- [x] performed a self-review of my code.
- [ ] commented on the particularly hard-to-understand areas of my code.
- [x] split my work into well-defined, bisectable commits, and I [named my commits well](https://github.com/dosbox-staging/dosbox-staging/blob/main/CONTRIBUTING.md#commit-messages).
- [x] applied the appropriate labels (bug, enhancement, refactoring, documentation, etc.)
- [x] [checked](https://github.com/dosbox-staging/dosbox-staging/blob/main/scripts/compile_commits.sh) that all my commits can be built.
- [ ] confirmed that my code does not cause performance regressions (e.g., by running the Quake benchmark).
- [ ] added unit tests where applicable to prove the correctness of my code and to avoid future regressions.
- [x] made corresponding changes to the documentation or the website according to the [documentation guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md).
- [x] [locally verified](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md#previewing-documentation-changes-locally) my website or documentation changes.

### CODE REVIEW

**Review by Grounded0 — APPROVED** [review]

**Comment by johnnovak** [comment]
Looks good @kcgen & nice test evidence! 🥇 

I'm wondering if it would be better to rephrase the sentence for macOS a bit. At the top of the Mac downloads page we're stating this:

> This package is compatible with macOS 10.15 (Catalina) or newer and supports both Intel and M1 Macs.

So saying "From the x86 family of processors, all x86-64 processors are supported" is slightly confusing for Mac users because:

- generally, they're not this technical; they understand Intel and M1 only because Apple marketing materials use these names,
- macOS is only supported on Intel (officially), not AMD,
- we also state that the earliest OS version you _might be able_ to run Staging on is Snow Leopard (I doubt it would actually work, or even be usable, but whatever...). Snow Leopard was the last macOS that still supported 32-bit Intel processors.

Given all that, probably this would be better to say:

> Both 64-bit Intel and M1 processors are supported.

**Comment by Grounded0** [comment]
@johnnovak 

Sounds good.

**Comment by kcgen** [comment]
Very succinctly worded; thanks @johnnovak and @Grounded0 .

Will push the revision tomorrow am..

**Comment by johnnovak** [comment]
> Sounds about right. I would rephrase as:
> 
> Both 64-bit Intel and Apple silicon processors are supported.

Right, I'm behind the times 😄 Let's go with this as you suggested:

> Both 64-bit Intel and Apple silicon processors are supported.

@kcgen ☝🏻

**Comment by kcgen** [comment]
all set for round two:

**Review by johnnovak — APPROVED** [review]

**Comment by Grounded0** [comment]
I have a remove snap link from linux.md PRs coming once this is deployed.

**Comment by kcgen** [comment]
Thanks for the double-check, @johnnovak.
Will merge.
